lib/spack/spack/build_environment.py

```diff
@@ -199,6 +200,13 @@ def set_compiler_environment_variables(pkg, env):
     env.set('SPACK_F77_RPATH_ARG', compiler.f77_rpath_arg)
     env.set('SPACK_FC_RPATH_ARG',  compiler.fc_rpath_arg)
 
+    # Set the target parameters that the compiler will add
+    isa_arg = ''
+    isa_target = spec.architecture.target.isa_target_for(compiler)
+    if compiler.isa_flag and isa_target:
+        isa_arg = '{0}={1}'.format(compiler.isa_flag, isa_target)
```

Comment:
Suggestion: should the json know what flags to add to different compilers too?  Seems like that would be useful.  Could it be part of the compiler mapping?  I am not suggesting we add everything Spack knows about compilers but I think this is one that clients of `archspec` will want to know.  Are there cases where the `isa_flag` differs across versions of the same compiler?

I guess the tension here is that Spack will still want its compiler classes to know the `isa_flag` so that builds can use it.  Still think it could be useful in `archspec`, though.

Reply:
> Are there cases where the isa_flag differs across versions of the same compiler?

There are cases like that, or at least we should expect them. For instance `-mcpu` and `-march` should be used depending whether the target is power or not. Which reminds me I need to polish the Power support :smiley:

Reply:
This part has been extensively refactored in e9beda6. Now the JSON file contains information on the optimization flags that needs to be used. Spack just delegates to `MicroArchitecture` objects to get the correct flags.